IFC_Engine/Query/GetIfcProperties.cs

```diff
@@ -0,0 +1,82 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Adapters.IFC.Properties;
+using BH.oM.Base;
+using BH.oM.Reflection.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Engine.Adapters.IFC
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+        
+        [Description("Retrieves IFC properties that are attached to a BHoM object. If a property with given name exists in both collections of pulled properties and the ones to push, the latter is returned.")]
+        [Input("bHoMObject", "BHoMObject to which the properties will be attached.")]
+        [Output("ifcProperties")]
+        public static List<IfcProperty> GetIfcProperties(this IBHoMObject bHoMObject)
+        {
+            if (bHoMObject == null)
+                return null;
+
+            Compute.UnsupportedUnitsWarning();
+
+            IfcPulledProperties pullFragment = bHoMObject.Fragments?.FirstOrDefault(x => x is IfcPulledProperties) as IfcPulledProperties;
+            IfcPropertiesToPush pushFragment = bHoMObject.Fragments?.FirstOrDefault(x => x is IfcPropertiesToPush) as IfcPropertiesToPush;
```

Comment:
This can be simplified:
```suggestion
            IfcPulledProperties pullFragment = bHoMObject.FindFragment<IfcPulledProperties>();
            IfcPropertiesToPush pushFragment = bHoMObject.FindFragment<IfcPropertiesToPush>();
```

remember to add `using BH.Engine.Base;`.